Week 1 นายยศพล ถิรพงศ์ชาติ 663380568-9

Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello, Python")

Hello, Python


Lab 1.2: Working with Variables and Basic Data Types

Declare and print variables

In [6]:
Age = 21
GPA = 3.33
Favorite_Color = "Red"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  21 <class 'int'>
GPA =  3.33 <class 'float'>
Favorite_Color =  Red <class 'str'>
Student =  True <class 'bool'>


Perform simple arithmetic

In [11]:
num1 = 13
num2 = 7
num3 = 14
num4 = 11
sum = num1 + num2
sum2 = sum * num3
sum3 = sum2 / num4
print("Sum1 = ",sum)
print("Sum2 = ",sum2)
print("Sum3 = ",sum3)

Sum1 =  20
Sum2 =  280
Sum3 =  25.454545454545453


tring concatenation and f-strings

In [12]:
First_name = "Nathanel"
Last_name = "jingjing"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Nathanel jingjing


User input and type conversion

In [15]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 5
type of A is :  <class 'str'>
5  + 5 =  10


Activity 1.1: Class Discussion - Uses of Scripting Format: Open discussion, possibly in small groups first, then full class. Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses." Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.

คำตอบ:การอ่านไฟล์ให้คำนวนการคาดการณ์หรือทำนายผล หรือ ทำการตรวจจับ ข่าวจริงเท็จได้

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


1.Word problems

In [ ]:
width = 5
length = 8

area = width * length

print("Width:", width)
print("Length:", length)
print("Area:", area, "square meters")

2.Puzzler

In [16]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))

13.5
<class 'float'>


ได้ผลลัพธ์: 13.5

Type: float

ทำไมถึงเป็น Float เพราะ เมื่อนำ int กับ Float มาบวกกัน จะแปลงType เป็น Float อัตโนมัติ

In [17]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old

Type : Str

ทำไมถึงเป็น str เพราะ age 30 ไม่ได้มี type เป็น int แต่เป็น str การใช้ + กับ type str เลยเป็นการ ต่อคำของประโยค

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

ได้ผลลัพธ์: False

Type : Bool

ทำไมถึงเป็น Bool เพราะ กำหนด True,False ที่เป็น type Bool

In [20]:
# Puzzler 4
# What happens here?
num_str = input("Enter a number: ")
result = int(num_str) + 5
print(result)

Enter a number: 5
10


ได้ผลลัพธ์: เลขที่กรอกบวกกับ5

Type : int

ทำไมถึงเป็น int เพราะได้มี การกำหนด num_str ให้เป็น int และบวกกับ 5 ที่เป็น int